### DISCUSSION TITLE
Error opening ExodusII mesh file

### DISCUSSION THREAD

**Discussion by xiaojbing** [idaholab/moose#19505] [Q&A Meshing] [answered]
I use pointwise to get the mesh. The MOOSE version is 2021-11-29, but it works well in the version 2020-07-16

```

EXODUS: Error: Attempting to open the netcdf-4 file:
	'/home/xjb/projects/cat/plane.e'
	with a netcdf library that does not support netcdf-4
Error opening ExodusII mesh file: /home/xjb/projects/cat/plane.e
Stack frames: 17
0: libMesh::print_trace(std::ostream&)
1: libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*)
2: libMesh::ExodusII_IO_Helper::open(char const*, bool)
3: libMesh::ExodusII_IO::read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
4: libMesh::NameBasedIO::read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
5: libMesh::UnstructuredMesh::read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, void*, bool, bool)
6: FileMesh::buildMesh()
7: MooseMesh::init()
8: SetupMeshAction::act()
9: Action::timedAct()
10: ActionWarehouse::executeActionsWithAction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
11: ActionWarehouse::executeAllActions()
12: MooseApp::runInputFile()
13: MooseApp::run()
14: ./cat-opt(+0x3153) [0x557900d69153]
15: __libc_start_main
16: ./cat-opt(+0x33dd) [0x557900d693dd]
[0] ../src/mesh/exodusII_io_helper.C, line 551, compiled Sep 20 2021 at 12:34:20


*** ERROR ***
Error opening ExodusII mesh file: /home/xjb/projects/cat/plane.e

Stack frames: 7
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: void mooseError<char const*>(char const*&&)
3: /home/xjb/projects/moose/framework/libmoose-opt.so.0(+0x78d054) [0x7f325842a054]
4: ./cat-opt(+0x3153) [0x557900d69153]
5: __libc_start_main
6: ./cat-opt(+0x33dd) [0x557900d693dd]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

```

**Comment by GiudGiud** [comment, marked as answer]
Hello

Libmesh needs to be configured with netcdf4 to load this file. 
Did you re-create the mesh file between the two versions? If you did, make sure to see https://github.com/idaholab/moose/discussions/19264 before re-exporting it. 

Guillaume

**Reply by xiaojbing** [reply]
No,I have't re-creat it. the mesh file between the two version is same.

**Reply by GiudGiud** [reply]
did you use conda for libmesh or did you install from the scripts in moose/scripts/?

**Reply by xiaojbing** [reply]
I used conda, just like the MOOSE website.

**Reply by GiudGiud** [reply]
@milljm conda build has libmesh compiled without HDF5 / netcfd4 right?

**Reply by GiudGiud** [reply]
@roystgnr for the sake of closing this, how do we enable netcdf4 in libmesh again? I remember searching at the time and could not find it

**Reply by roystgnr** [reply]
NetCDF4 is automatically enabled whenever HDF5 is enabled, but HDF5 is never enabled automatically in libMesh (it's one of those things that often is preinstalled in /usr/lib/ but we never want to auto-include that whole directory lest we get whatever atavistic package versions are there).

If you already have HDF5, setting `$HDF5_DIR` is the best way to get us to check `$HDF5_DIR/libs` and `$HDF5_DIR/include` for libraries and headers.  If you have an install but it doesn't have that subdirectory structure, making a couple symbolic links in your own new `$HDF5_DIR` is currently the best way to handle it.

If you don't already have HDF5, you ought to be able to get it now just by using Moose's `update_and_rebuild_petsc.sh`, which will ask PETSc to build HDF5 if it doesn't find it elsewhere, after which point libMesh will autodetect PETSc's HDF5 installation.
